tests/autobahn/test_autobahn.py

```diff
@@ -0,0 +1,131 @@
+import json
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+from typing import Any, Dict, Generator, List
+
+import pytest
+from pytest import TempPathFactory
+from python_on_whales import docker
+
+
+@pytest.fixture(scope="session")
+def report_dir(tmp_path_factory: TempPathFactory) -> Path:
+    return tmp_path_factory.mktemp("reports")
+
+
+@pytest.fixture(scope="session", autouse=True)
+def build_autobahn_testsuite() -> Generator[None, None, None]:
+    if shutil.which("docker") is None:
+        pytest.skip(msg="The docker binary file was not found.")
+    else:
+        docker.build(
+            file="tests/autobahn/Dockerfile.autobahn",
+            tags=["autobahn-testsuite"],
+            context_path=".",
+        )
+        try:
+            yield
+        finally:
+            docker.image.remove(x="autobahn-testsuite")
```

Comment:
Skip interrupts the test so there's no need to indent this block.
```suggestion

    docker.build(
        file="tests/autobahn/Dockerfile.autobahn",
        tags=["autobahn-testsuite"],
        context_path=".",
    )
    try:
        yield
    finally:
        docker.image.remove(x="autobahn-testsuite")
```